prefect_great_expectations/validation.py

```diff
@@ -0,0 +1,102 @@
+"""Tasks for performing Great Expectations validations"""
+
+from pathlib import Path
+from typing import Dict, Optional, Union
+
+from great_expectations import DataContext
+from great_expectations.checkpoint import Checkpoint
+from great_expectations.checkpoint.types.checkpoint_result import CheckpointResult
+from prefect import get_run_logger, task
+
+
+class GreatExpectationValidationError(Exception):
+    """
+    Signals that a task failed due to a failed Great
+    Expectations validation.
+
+    Args:
+        result: A CheckpointResult containing details
+            of the failed validation.
+    """
+
+    def __init__(self, result: CheckpointResult):
+        self.result = result
+        super().__init__(
+            "Great Expectations Validation failed. "
+            "Check result on this exception for more details."
+        )
+
+
+@task
+def run_checkpoint_validation(
+    run_name: Optional[str] = None,
+    checkpoint_name: Optional[str] = None,
+    checkpoint: Optional[Checkpoint] = None,
+    checkpoint_kwargs: Optional[Dict] = None,
+    data_context_root_dir: Optional[Union[str, Path]] = None,
+    data_context: Optional[DataContext] = None,
+    runtime_environment: Optional[Dict] = None,
+    raise_on_validation_failure: bool = True,
+):
+    """
+    Task that performs a Great Expectations validation based on the provided checkpoint
+        and data context
+
+    Args:
+        run_name: The name of the Great Expectations validation run. Defaults to
+            timestamp if not provided.
+        checkpoint_name: The name of the Checkpoint to use for validation.
+        checkpoint: A Checkpoint object to use for validation. Overrides
+            `checkpoint_name` if both are provided.
+        checkpoint_kwargs:  A dictionary with values used to provide configuration to
+            the task's Checkpint at runtime. Keys should match the parameters of
+            `CheckpointConfig`.
+        data_context_root_dir: Path to the great_expectations directory.
+        data_context: A DataContext object to use during validation. Overrides
+            `data_context_root_dir` if both are provided.
+        runtime_environment: A dictionary with values to overwrite config in
+            `great_expectations.yml` at run time.
+        raise_on_validation_failure: If `True`, the task will raise a
+            GreatExpectationValidationError when validation fails. If `False`,
+            the task will return the result of the validation.
+
+    Raises:
+        GreatExpectationValidationError: Signals that a GE validation failed.
+            Details of the failure can be found by inspecting the `result`
+            attribute of the exception.
+
+    Returns:
+        CheckpointResult: Detailed result of the validation run in the task.
+    """
+    logger = get_run_logger()
+
+    logger.info("Running Great Expectations validation...")
+
+    runtime_environment = runtime_environment or dict()
```

Comment:
I think `{}` may be more canonical; not entirely sure.
```suggestion
    runtime_environment = runtime_environment or {}
```